ownCloud/Client/Actions/Scanner/ScanViewController.swift

```diff
@@ -0,0 +1,526 @@
+		saveSection.add(row: fileNameRow!)
+		self.addSection(saveSection)
+
+		// Options section
+		// - Format
+		formatSegmentedControl = UISegmentedControl(items: availableExportFormats.map({ (format) in return format.name }))
+		formatSegmentedControl?.selectedSegmentIndex = 0
+		formatSegmentedControl?.isUserInteractionEnabled = true
+		formatSegmentedControl?.addTarget(self, action: #selector(updateExportFormat), for: .valueChanged)
+		optionsSection.add(row: StaticTableViewRow(label: "File format".localized, alignment: .left, accessoryView: formatSegmentedControl, identifier: "format"))
+
+		// - One file per page
+		oneFilePerPageRow = StaticTableViewRow(switchWithAction: nil, title: "Create one file per page".localized, value: false, identifier: "one-file-per-page")
+		optionsSection.add(row: oneFilePerPageRow!)
+		self.addSection(optionsSection)
+
+		updateExportFormat()
+	}
+
+	required init?(coder: NSCoder) {
+		fatalError("init(coder:) has not been implemented")
+	}
+
+	@objc func updateExportFormat() {
+		if let selectedFormatIndx = formatSegmentedControl?.selectedSegmentIndex {
+			self.exportFormat = availableExportFormats[selectedFormatIndx]
+		}
+	}
+
+	@objc func cancel() {
+		self.dismiss(animated: true, completion: nil)
+	}
+
+	@objc func save() {
+		if let fileName = fileNameRow?.value as? String, let pages = pagesCollectionViewController?.pages, let exporter = self.exportFormat?.exporter, let core = core {
+			let progressHUDViewController = ProgressHUDViewController(on: self, label: "Saving".localized)
+
+			DispatchQueue.global(qos: .userInitiated).async {
+				let tmpURL = URL(fileURLWithPath: NSTemporaryDirectory()).appendingPathComponent(UUID().uuidString)
+
+				try? FileManager.default.createDirectory(at: tmpURL, withIntermediateDirectories: true, attributes: nil)
+
+				var exportedURLs : [URL]?
+
+				if self.exportFormat?.supportsMultiPage == true, (self.oneFilePerPageRow?.value as? Bool) == true {
+					for page in pages {
+						if let exportedPageURLs = exporter(tmpURL, [page]) {
+							if exportedURLs == nil {
+								exportedURLs = []
+							}
+							exportedURLs?.append(contentsOf: exportedPageURLs)
+						} else {
+							exportedURLs = nil
+							break
+						}
+					}
+				} else {
+					exportedURLs = exporter(tmpURL, pages)
+				}
+
+				Log.log("ExportPDF=\(String(describing: exportedURLs))")
+
+				if let targetItem = self.targetFolderItem, let exportedURLs = exportedURLs {
+					for exportedURL in exportedURLs {
+						core.importFileNamed(fileName, at: targetItem, from: exportedURL, isSecurityScoped: false, options: [ .automaticConflictResolutionNameStyle : OCCoreDuplicateNameStyle.bracketed.rawValue ], placeholderCompletionHandler: nil, resultHandler: nil)
```

Comment:
@felix-schwarz error handling in comletion handlers missing?

Reply:
If an error occurs, the SDK will generate an issue and tell the app to present it.